### PR TITLE
fix: null value not persisted for properties of type JSON, Any, or Ob…

### PR DESCRIPTION
**Pull request by siimsams**
Fix the following issue on latest master.

fixes https://github.com/loopbackio/loopback-datasource-juggler/issues/1895

## Checklist

- [X] [Sign off your commits](https://loopback.io/doc/en/contrib/code-contrib.html) with DCO (Developer Certificate of Origin)
- [X] `npm test` passes on your machine
- [X] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [X] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by siimsams** [comment]
I have added tests aswell. Please let me know if there is anything missing here. I'd really like to get this merged as we have many projects with this issue.

**Comment by ewrayjohnson** [comment]
@siimsams: Thanks.  Although I proposed the fix, I had not found the time to create the tests.

**Comment by siimsams** [comment]
> @siimsams: Thanks. Although I proposed the fix, I had not found the time to create the tests.

Thank you for proposing the fix. I added the tests aswell. Hope this gets approved and merged. We want to use latest versions but this issue blocks us from doing that.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "loopback-datasource-juggler",
-  "version": "4.28.6",
```
[the remark this answers is not in the document]
Comment (siimsams) [reply package.json]: Oh, i did not know. Fixed it.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+2023-06-19, Version 4.28.7
```
[the remark this answers is not in the document]
Comment (siimsams) [reply CHANGES.md]: Oh, i did not know. Fixed it.

**Comment by siimsams** [comment]
Fixed the commit message. It would be nice to have these required prefixes detailed in the pr/commit guidelines.

**Comment by dhmlau** [comment]
@siimsams, there's still error from the commit linter
```
   type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
```

I suggest to squash the commit and have the commit message something like what you have in the title (make sure it's within 50 characters).

**Review by dhmlau — APPROVED** [review]
Your changes LGTM. Let's get the commit linting error fixed and we can go ahead to merge. Thanks.

**Comment by siimsams** [comment]
Squashed everything. Should be good to go now. 

> Your changes LGTM. Let's get the commit linting error fixed and we can go ahead to merge. Thanks.

**Review by dhmlau — APPROVED** [review]

**Comment by dhmlau** [comment]
@siimsams, thanks for your contribution. Your PR has landed! 🎉

**Comment by siimsams** [comment]
Thank you so much!

**Pull request merged**